src/Integrations/Clearbit/SettingsClearbit.php

```diff
@@ -0,0 +1,456 @@
+	{
+		$this->clearbitClient = $clearbitClient;
+	}
+
+	/**
+	 * Register all the hooks
+	 *
+	 * @return void
+	 */
+	public function register(): void
+	{
+		\add_filter(self::FILTER_SETTINGS_SIDEBAR_NAME, [$this, 'getSettingsSidebar']);
+		\add_filter(self::FILTER_SETTINGS_GLOBAL_NAME, [$this, 'getSettingsGlobalData']);
+		\add_filter(self::FILTER_SETTINGS_IS_VALID_NAME, [$this, 'isSettingsValid'], 10, 2);
+		\add_filter(self::FILTER_SETTINGS_GLOBAL_IS_VALID_NAME, [$this, 'isSettingsGlobalValid']);
+	}
+
+	/**
+	 * Determine if settings are valid.
+	 *
+	 * @param string $formId Form ID.
+	 * @param string $type Integration type.
+	 *
+	 * @return boolean
+	 */
+	public function isSettingsValid(string $formId, string $type): bool
+	{
+		if (!$this->isSettingsGlobalValid()) {
+			return false;
+		}
+
+		$typeItems = Filters::ALL[self::SETTINGS_TYPE_KEY]['integration'];
+
+		if (!isset($typeItems[$type])) {
+			return false;
+		}
+
+		$useClearbit = $this->getSettingsValue($typeItems[$type]['use'], $formId);
+
+		if (empty($useClearbit)) {
+			return false;
+		}
+
+		$emailSet = $this->getSettingsValue($typeItems[$type]['email'], $formId);
+
+		if (empty($emailSet)) {
+			return false;
+		}
+
+		$mapSet = $this->getOptionValueGroup($typeItems[$type]['map']);
+
+		if (empty($mapSet)) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Determine if settings global are valid.
+	 *
+	 * @return boolean
+	 */
+	public function isSettingsGlobalValid(): bool
+	{
+		$isUsed = $this->isCheckboxOptionChecked(self::SETTINGS_CLEARBIT_USE_KEY, self::SETTINGS_CLEARBIT_USE_KEY);
+		$apiKey = !empty(Variables::getApiKeyClearbit()) ? Variables::getApiKeyClearbit() : $this->getOptionValue(self::SETTINGS_CLEARBIT_API_KEY_KEY);
+
+		if (!$isUsed || empty($apiKey)) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get Settings sidebar data.
+	 *
+	 * @return array<string, mixed>
+	 */
+	public function getSettingsSidebar(): array
+	{
+		return [
+			'label' => __('Clearbit', 'eightshift-forms'),
+			'value' => self::SETTINGS_TYPE_KEY,
+			'icon' => Filters::ALL[self::SETTINGS_TYPE_KEY]['icon'],
+		];
+	}
+
+	/**
+	 * Get Form settings data array
+	 *
+	 * @param string $formId Form Id.
+	 *
+	 * @return array<int, array<string, mixed>>
+	 */
+	public function getSettingsData(string $formId): array
+	{
+		return [];
+	}
+
+	/**
+	 * Get global settings array for building settings page.
+	 *
+	 * @return array<int, array<string, mixed>>
+	 */
+	public function getSettingsGlobalData(): array
+	{
+		$isUsed = $this->isCheckboxOptionChecked(self::SETTINGS_CLEARBIT_USE_KEY, self::SETTINGS_CLEARBIT_USE_KEY);
+
+		$output = [
+			[
+				'component' => 'intro',
+				'introTitle' => __('Clearbit', 'eightshift-forms'),
+			],
+			[
+				'component' => 'intro',
+				'introTitle' => __('How to get the API key?', 'eightshift-forms'),
+				'introTitleSize' => 'small',
+				// phpcs:ignore WordPress.WP.I18n.NoHtmlWrappedStrings
+				'introSubtitle' => __('<ol>
```

Comment:
I'm not overly fond of seeing HTML in a translatable string. `sprintf` would be a better match here IMO. That way, you can have HTML outside of the translatable strings.